torch/_dynamo/side_effects.py

```diff
@@ -836,3 +850,13 @@ def allow_side_effects_under_checkpoint(tx: "InstructionTranslator"):  # type: i
         yield
     finally:
         tx.output.current_tracer.allow_side_effects_under_checkpoint = orig_val
+
+
+@contextlib.contextmanager
+def disallow_side_effects_under_generator(tx: "InstructionTranslator"):  # type: ignore[name-defined]  # noqa: F821
```

Comment:
I'm assuming this is because `InstructionTranslator` needs to be imported in a TYPE_CHECKING block